kratos/includes/define.h

```diff
@@ -484,6 +484,142 @@ catch(...) { Block KRATOS_THROW_ERROR(std::runtime_error, "Unknown error", MoreI
     KRATOS_REGISTER_VARIABLE(name##_ZY) \
     KRATOS_REGISTER_VARIABLE(name##_ZZ)
 
+//-----------------------------------------------------------------
+//
+//  Variables time derivatives
+//
+//-----------------------------------------------------------------
+
+#ifdef KRATOS_REGISTER_VARIABLE_TIME_DERIVATIVE
+#undef KRATOS_REGISTER_VARIABLE_TIME_DERIVATIVE
+#endif
+#define KRATOS_REGISTER_VARIABLE_TIME_DERIVATIVE(variable, variable_derivative) \
+    AddVariableTimeDerivative(variable, variable_derivative); \
```

Comment:
If we do this, we are forcing this at registration time, meaningfor example that VELOCITY will be always the first derivative of, say, displacement.

I am not sure if this is too rigid (in the sense that maybe there are already conflictive usages for some variable)

As a completely different comment, why do we need this to be a macro?

Reply:
I defined a macro in order to be consistent with the KratosComponents I used as base idea. The thing is, when velocity is not the derived of displacement?, do you mean something like MESH_VELOCITY?, please I don't know what is the potential problem. Besides, the idea of define it in compiling time is for performance (as the ideal would be in order to define a generic explicit scheme like the one @rubenzorrilla  is developing, which requires a performance behaviour in order to don't be a bottleneck in the explicit simulation)